crates/solvers/src/util/conv.rs

```diff
@@ -0,0 +1,32 @@
+//! Conversion utilities.
+
+use crate::domain::eth;
+use bigdecimal::BigDecimal;
+use ethereum_types::U256;
+
+/// Converts a `BigDecimal` value to a `eth::Rational` value.
+pub fn decimal_to_rational(d: &BigDecimal) -> Option<eth::Rational> {
```

Comment:
It is unclear under what conditions this function returns None. eth::Rational is a ratio of U256. Does this mean that all values >= eth::Rational::MIN and <= eth::Rational::MAX should return Some and all other values None?

I tried to convert `BigDecimal::from_str("1.1111111111111111111111111111111111111111111111111111111111111111111111111111111``")` which failed.

This feels like a function that should have a unit test to me too because the implementation is pretty difficult and the previous criticism of unit tests with regards to mocking and making refactoring harder doesn't apply.

Reply:
Will add some unit tests here.

Reply:
> tried to convert `BigDecimal::from_str("1.1111111111111111111111111111111111111111111111111111111111111111111111111111111")` which failed.

I thought it panic'ed - phew! This is expected to return `None` is it would overflow the numerator (and, incidentally, the denominator as well). Added a unit test and refactored the implementation to make sure it reduces the ratio before returning.

Reply:
45836eb5a6db76e0b9f942fb01e4cac6a1b9b1f2